src/Forms/Tab.php

```diff
@@ -99,6 +98,14 @@ public function setID($id)
         return $this;
     }
 
+    public function setName($name)
+    {
+        if ($this->ID() === Convert::raw2htmlid($this->getName())) {
+            $this->setID(Convert::raw2htmlid($name));
+        }
+        return parent::setName($name);
+    }
```

Comment:
This ensures the `id` attribute in the HTML markup correctly reflects the name, even if the name is set after the tab was created. This is needed by https://github.com/silverstripe/silverstripe-userforms/pull/1315/files#diff-d659fc65ecb263c36d5428815131e80ef3c1750bf18e18860f1fd9d79d4df3e0R195 because the relation name doesn't match the name we want for the tab.

It's also just a sensible change to make.

Note that we don't set the ID if the ID was manually set to something other than the name.